rainier-core/src/main/scala/com/stripe/rainier/core/Likelihood.scala

```diff
@@ -4,42 +4,68 @@ import com.stripe.rainier.compute._
 import scala.collection.mutable.ArrayBuffer
 
 trait Likelihood[T] {
-  private[core] type P
-  private[core] def wrapping: Wrapping[T, P]
-  private[core] def logDensity(value: P): Real
+  def real: Real
+  def placeholders: List[Variable]
+  def extract(t: T): List[Double]
 
-  def target(value: T): Target =
-    Target(logDensity(wrapping.wrap(value)))
-
-  def sequence(seq: Seq[T]): Target = {
-    val ph = wrapping.placeholder(seq)
-    val real = logDensity(ph.value)
-    val variables = ph.variables(Nil)
+  def fit(seq: Seq[T]): RandomVariable[Unit] = {
     val arrayBufs =
-      variables.map { _ =>
+      placeholders.map { _ =>
         new ArrayBuffer[Double]
       }
     seq.foreach { t =>
-      val doubles = ph.extract(t, Nil)
+      val doubles = extract(t)
       arrayBufs.zip(doubles).foreach {
         case (a, d) => a += d
       }
     }
     val placeholdersMap =
-      variables.zip(arrayBufs.map(_.toArray)).toMap
-    new Target(real, placeholdersMap)
+      placeholders.zip(arrayBufs.map(_.toArray)).toMap
```

Comment:
I'm on board with these pretty minor changes, but I still don't totally understand why this works this way. I'm imagining that T is (Double, Double) so for a given pair of Doubles, placeholders is a list of two variables and extract gives us the list with those two doubles. Then, given a seq of pairs of doubles, `arrayBufs` is a list of two ArrayBuffers and for each pair in our list, we extract the list of Doubles and zip and add, then put that in placeholdersMap which goes into a new RandomVariables's `targets` so morally these terms are getting added to the density. You mentioned that this is a chunk of the compute graph. Is the idea that some other chunk might also fit to the same observations? Or are the observations supposed to get spread around?
Also, my understanding is that now, `Likelihoods` are only used for computing likelihoods in MCMC and are totally decoupled from Distributions in general so you can make whatever likelihood you want, but can also use `Distribution.likelihood` to make a `Likelihood` out of a `Distribution`. That's nice.

Reply:
> You mentioned that this is a chunk of the compute graph. Is the idea that some other chunk might also fit to the same observations? Or are the observations supposed to get spread around?

It's a chunk of the compute graph in that it represents the contribution to the overall joint density of this single likelihood function; but there are also all the priors on the parameters that contribute, and then there may also be other observations, with some different structure, that get fit separately.

Does that answer your question?